pkg/operator/controller/controller.go

```diff
@@ -336,42 +352,49 @@ func (r *reconciler) ensureRouterForIngress(ci *ingressv1alpha1.ClusterIngress,
 		Controller: &trueVar,
 	}
 
-	lbService, err := r.ensureLoadBalancerService(ci, infraConfig, current)
+	errs := []error{}
+	lbService, err := r.ensureLoadBalancerService(ci, current)
 	if err != nil {
-		return fmt.Errorf("failed to ensure load balancer service for %s: %v", ci.Name, err)
+		errs = append(errs, fmt.Errorf("failed to ensure load balancer service for %s: %v", ci.Name, err))
 	}
-
-	err = r.ensureDNS(ci, dnsConfig, lbService)
-	if err != nil {
-		return fmt.Errorf("failed to ensure DNS for %s: %v", ci.Name, err)
+	if err = r.ensureDNS(ci, lbService); err != nil {
+		errs = append(errs, fmt.Errorf("failed to ensure DNS for %s: %v", ci.Name, err))
 	}
 
 	internalSvc, err := r.ensureInternalRouterServiceForIngress(ci, deploymentRef)
 	if err != nil {
-		return fmt.Errorf("failed to create internal router service for clusteringress %s: %v", ci.Name, err)
+		errs = append(errs, fmt.Errorf("failed to create internal router service for clusteringress %s: %v", ci.Name, err))
+		return utilerrors.NewAggregate(errs)
```

Comment:
I thought the idea not to return on non-fatal errors.  Can't we append the error value and move on?

Reply:
Internal service creation has no external dependency, so this will fail when api service fails or service IP is already taken, etc. If api service fails then everything should fail so I categorized this as fatal error. Are you considering transient api failures?

Reply:
We want to move in the direction of trying to do as many tasks as we can even as some tasks fail, but we don't necessarily need to do that now.  And you're right, and I get what you're saying—the issue we urgently must fix is that a non-transient error can prevent the operator from performing subsequent tasks, and you are fixing that problem here.